rslib/src/image_occlusion/imagedata.rs

```diff
@@ -17,57 +15,138 @@ impl Collection {
+        let image_file = format!("<img src='{}'></img>", &image_filename);
+
+        let note_type_name = "Image Occlusion - Anki Ecosystem";
+        let add_note_types = self
+            .add_io_note_type(&note_type_name.to_string(), &card_template_json)?;
+        
+        let io_notes = self.get_io_notes(notes_data);
         
-        let ans_image_name = [&rand_str, "ans.svg"].join("-");
-        let actual_ans_name_after_adding =
-            mgr.add_file(&mut ctx, &ans_image_name, ans_image_data)?;
-    
         let nt = self
-            .get_notetype_by_name("Basic")?
+            .get_notetype_by_name(&note_type_name)?
             .ok_or(AnkiError::NotFound)?;
+    
+        for id in random_id_list {
+            let mut note = nt.new_note();
+            note.set_field(0, id)?;
+            note.set_field(1, io_notes[0].clone())?;
+            note.set_field(2, &image_file)?;
+            note.set_field(3, &svg_mask)?;
+            note.set_field(4, io_notes[1].clone())?;
+            note.set_field(5, io_notes[2].clone())?;
+            note.set_field(6, io_notes[3].clone())?;
+            note.set_field(7, io_notes[4].clone())?;
+            note.set_field(8, io_notes[5].clone())?;
 
-        let mut note = nt.new_note();
-        note.set_field(0, "some text")?;
+            // the routine should return these changes
+            let changes = self.add_note(&mut note, DeckId(1))?;
+        }
+        
+        resul_add_notes.is_added = true;
+        Ok(resul_add_notes)
+    }
+
+    pub fn get_data_using_key(&mut self, json: &Value, key: &str) -> String {
+        let data = json.get(key);
+        if data != None {
+            data.unwrap().as_str().unwrap().to_string()
+        } else {
+            "".to_string()
+        }
+    }
 
-        // the routine should return these changes
-        let changes = self.add_note(&mut note, DeckId(1))?;
+    pub fn get_io_note_field(&mut self) -> Vec<NoteField> {
+        let io_fields = vec![
+            NoteField::new("ID (hidden)"), 
+            NoteField::new("Header"), 
+            NoteField::new("Question Mask"), 
+            NoteField::new("Footer"), 
+            NoteField::new("Sources"), 
+            NoteField::new("Remarks"), 
+            NoteField::new("Extra 1"), 
+            NoteField::new("Extra 2")
+        ];
+        return io_fields;
+    }
 
-        resul_add_notes.is_added = false;
-        Ok(resul_add_notes)
+    pub fn get_io_card_templates(&mut self, template_name: &String, card_template_json: &Value) -> Vec<CardTemplate> {
+        let io_qfmt = self.get_data_using_key(&card_template_json, "qfmt");
+        let io_afmt = self.get_data_using_key(&card_template_json, "afmt");
+        let io_css = self.get_data_using_key(&card_template_json, "css");
+        
+        println!("{}\n\n{} \n\n{}", template_name, io_qfmt, io_afmt);
+        
+        let io_templates: CardTemplate = CardTemplate::new(template_name, io_qfmt, io_afmt);
```

Comment:
@dae Please help with this errors
```
Traceback (most recent call last):
  File "/home/mani/.cache/bazel/_bazel_mani/c32e553acd1c32076f1c9e723be5100c/execroot/ankidesktop/bazel-out/k8-fastbuild/bin/qt/runanki.runfiles/ankidesktop/qt/aqt/mediasrv.py", line 528, in wrapped
    if data := handler():
  File "/home/mani/.cache/bazel/_bazel_mani/c32e553acd1c32076f1c9e723be5100c/execroot/ankidesktop/bazel-out/k8-fastbuild/bin/qt/runanki.runfiles/ankidesktop/qt/aqt/mediasrv.py", line 509, in <lambda>
    return lambda: getattr(aqt.mw.col._backend, f"{endpoint}_raw")(request.data)
  File "/home/mani/.cache/bazel/_bazel_mani/c32e553acd1c32076f1c9e723be5100c/execroot/ankidesktop/bazel-out/k8-fastbuild/bin/qt/runanki.runfiles/ankidesktop/pylib/anki/_backend/generated.py", line 2006, in add_image_occlusion_notes_raw
    return self._run_command(17, 1, message)
  File "/home/mani/.cache/bazel/_bazel_mani/c32e553acd1c32076f1c9e723be5100c/execroot/ankidesktop/bazel-out/k8-fastbuild/bin/qt/runanki.runfiles/ankidesktop/pylib/anki/_backend/__init__.py", line 146, in _run_command
    raise backend_exception_to_pylib(err)
anki.errors.CardTypeError: Card template ⁨1⁩ in notetype '⁨Image Occlusion - Anki Ecosystem⁩' has a problem.<br>See the preview for more information.
```

Reply:
You're referencing {{Image}} but not defining an Image field

Reply:
(while for prototyping it's fine, in the long run we'll want to define the notetype in stock.rs instead of in TypeScript)

Reply:
I will move the templates to stock.rs file. Thanks